torch/onnx/_internal/fx/export.py

```diff
@@ -541,3 +655,253 @@ def compile(self, gm: torch.fx.GraphModule, _):
         decomposition_table=_ONNX_FRIENDLY_DECOMPOSITION_TABLE,
         use_binary_format=use_binary_format,
     )
+
+
+def _move_placeholder_to_front(graph_module: torch.fx.GraphModule) -> None:
+    """
+    This function move all placeholder nodes to the front of the graph node list.
+    In torch.fx.Graph, placehoder is a special assignment node. If it's not
+    executed in the beginning, it could overwrite values computed by upstream
+    nodes.
+    """
+
+    graph = graph_module.graph
+    placeholders = []
+    first_not_placeholder = None
+    for node in graph.nodes:
+        if node.op == "placeholder":
+            placeholders.append(node)
+        if first_not_placeholder is None and node.op != "placeholder":
+            first_not_placeholder = node
+    if first_not_placeholder is None:
+        return
+    for placeholder in placeholders:
+        first_not_placeholder.prepend(placeholder)
+
+
+def _replace_get_attr_with_placeholder(graph_module: torch.fx.GraphModule):
```

Comment:
```suggestion
def _replace_get_attr_with_placeholder(graph_module: torch.fx.GraphModule) -> List[torch.Tensor]:
```
Is this accurate?

Reply:
Yes. But I decide to use `Tuple[torch.Tensor, ...]` because it's not mutable. Thanks. :)